cli/cmd/cluster/cluster.go

```diff
@@ -171,10 +173,11 @@ type controlplaneUpdater struct {
 	ex            terraform.Executor
 }
 
-func (c controlplaneUpdater) getControlplaneChart(name string) (*chart.Chart, error) {
-	chart, err := platform.ControlPlaneChart(name)
+func (c controlplaneUpdater) getControlplaneChart(name, namespace string) (*chart.Chart, error) {
+	path := filepath.Join(c.assetDir, "cluster-assets", "charts", namespace, name)
```

Comment:
What happens if there's no assets directory present? Does this run after it has been recreated from Terraform state?

Reply:
Read the commit message :wink: 

```
    If controlplane chart is not installed, we attempt to re-install it from
    cluster assets directory. If chart is not found there, we bail, as we
    have no way of knowing if binary we run right now is the same binary
    which was previously used to deploy a cluster and we don't want to
    trigger cluster upgrade until we make sure that existing cluster is
    healthy.

    In such case, user should clone 'lokomotive' repository, copy the right
    version of the Helm chart into their assets directory and then re-run
    'lokoctl cluster apply'. This will change when we decouple Lokomotive
    release from 'lokoctl' binary, as then we will have access to the right
    Helm charts.
```

Also in such case, one can use `--skip-pre-update-health-check` flag.

Reply:
This seems pretty annoying, can we have some command/list of commands the user need to run documented.

Reply:
Right. Though is rather unlikely that user hits this. I think I'll create a "troubleshooting" section in the docs describing this.

Reply:
Added a documentation for it: https://github.com/kinvolk/lokomotive/pull/1181/commits/a139f8ee0b1c2bf15554832bf0549879470a80a3.